### DISCUSSION TITLE
404 not found error

### DISCUSSION THREAD

**Discussion by mamob07** [firefly-iii/firefly-iii#9411] [Firefly III] [answered]
### Support guidelines

- [x] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [x] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem
- [x] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

When I surf to firefly-iii in /public directory i get redirected to /public/login and not /public/index.php/login. And all the other things such as images and js scripts aren't loaded because of that.

### Debug information

Debug information generated at 2024-10-29 11:15:17 for Firefly III version **v6.1.19**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.19 / v2.1.0 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.12 (64bits) / fpm-fcgi / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>52428800 (50 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, sqlite, </td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Amsterdam + Europe/Rome</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, daily / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>it_IT + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td></td></tr><tr><td>Session start</td><td></td></tr><tr><td>Session end</td><td></td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>it_IT</td></tr><tr><td>User locale</td><td>it_IT</td></tr><tr><td>Locale(s) supported</td><td>it_IT.utf8: :x:<br>it_IT.UTF-8: :x:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Mobile Safari/537.36</td></tr></tbody></table>

### Expected behaviour

I expect all /public/<URL> URLs to be rewritten as /public/index.php/<URL>.

### Steps to reproduce

1. Surf to /public directory

### Additional info

This is the nginx_app.conf file:
```
# prevent HTTPoxy vulnerability
fastcgi_param HTTP_PROXY "";

location / {
    if ($http_x_forwarded_proto != "https") {
        return 301 https://$host$request_uri;
    }
    # try to serve file directly, fallback to rewrite
    try_files $uri @rewriteapp;
}

location @rewriteapp {
    # rewrite all to index.php
    rewrite ^(.*)$ /index.php$1 last;
}
```

**Comment by JC5** [comment]
Hi, thanks for opening an issue.

This is a configuration issue in the first place. Please try to configure your webserver so that `/` equals `/public`.

**Reply by mamob07** [reply]
I did it and now the error is that /login isn't rewrited to /index.php/login

**Reply by JC5** [reply, marked as answer]
Check out this nginx configuration:

https://laravel.com/docs/11.x/deployment#nginx
